chit_chat/serializers.py

```diff
@@ -55,3 +56,7 @@ def validate_members(self, users):
             raise exceptions.ValidationError('Must contain at least one user other than the requestor in this list.')
         non_requestor_users.append(requestor)
         return users
+
+    def create(self, validated_data):
+        member_pks = [member.pk for member in validated_data['members']]
+        return Room.objects.filter(members__in=member_pks).annotate(member_count=Count('members')).filter(member_count=len(member_pks)).first() or super().create(validated_data)
```

Comment:
I have a feeling that "or ..." at the end will get someone some day, but I've already nitpicked this a bit so up to you if ya want to change it .. maybe something like

```py
room = Room.objects.filter(members__in=member_pks).annotate(member_count=Count('members')).filter(member_count=len(member_pks)).first()

return room or super().create(validated_data)
```

Reply:
Great suggestion.